lib/marisa/agent.cc

```diff
@@ -4,8 +4,23 @@
 #include "marisa/agent.h"
 #include "marisa/grimoire/trie.h"
 #include "marisa/grimoire/trie/state.h"
+#include "marisa/key.h"
 
 namespace marisa {
+namespace {
+void UpdateAgentAfterCopyingState(const grimoire::trie::State &state,
+                                  Agent &agent) {
+  // Point the agent's key to the newly copied buffer if necessary.
```

Comment:
Can you explain a bit more? Why are these the only two status codes requiring a copy?

Reply:
I've now added more comments to the code explaining this